projects/onto-search/src/lib/onto-search-facet/date-range-facet/date-range-facet.component.ts

```diff
@@ -121,36 +123,52 @@ export class DateRangeFacetComponent extends OntoSearchFacetComponent implements
     }
   }
 
-  private getCalendarPanelButtons(): any {
-    return document.querySelectorAll('mat-calendar .mat-calendar-body-cell, mat-calendar button, mat-calendar .mat-icon-button');
+  private updateDayStyles(): void {
+    if (document.querySelector('mat-month-view')) {
+      this.addTooltipsToCalendarCells('mat-month-view', this.daysMap, DATE_PATTERNS.DAY);
+    } else if (document.querySelector('mat-multi-year-view')) {
+      this.addTooltipsToCalendarCells('mat-multi-year-view', this.yearsMap, DATE_PATTERNS.YEAR);
+    } else if (document.querySelector('mat-year-view')) {
+      this.addTooltipsToCalendarCells('mat-year-view', this.monthsMap, DATE_PATTERNS.MONTH);
+    }
   }
 
-  private subscribeToBtnListeners(buttons: any): void {
-    buttons.forEach((btn) => {
-      const unsubscribe = this.renderer.listen(btn, 'click', () => {
-        this.calendarOpened();
+  private addTooltipsToCalendarCells(view: string, datesMap: Map<string, number>, dateFormat: string): void {
+    const calendarCells = this.getCalendarCellsForView(view);
+    calendarCells.forEach((calendarCell) => {
+      datesMap.forEach((count, dateString) => {
+        const calendarCellDateString = moment(calendarCell.getAttribute('aria-label'), dateFormat);
+        if (calendarCellDateString.isSame(dateString)) {
+          this.createTooltipForCalendarCell(calendarCell, count);
+        }
       });
-      this.buttonUnsubscribeFns.push(unsubscribe);
     });
   }
 
-  private unsubscribeBtnListeners(): void {
-    this.buttonUnsubscribeFns.forEach((unsubscribeFn) => {
-      unsubscribeFn();
-    });
-    this.buttonUnsubscribeFns = [];
-  }
+  /**
+   * Creates a portal outlet from the calendar cell div and attaches the tooltip template from view
+   * Applies "onto-date-range-facet-tooltip" class to calendar cell child div
+   *
+   * @param calendarCell - calendar cell to which the span is appended
+   * @param toolTipContext - context for the tooltip template
+   */
+  private createTooltipForCalendarCell(calendarCell, toolTipContext: number): void {
+    const div = this.getCalendarCellChildDiv(calendarCell);
```

Comment:
Well. It's good that you have hidden the DOM operation in these methods. But, why do you leak implementation details in the method names :) ?!